### ISSUE TITLE
Is it possible/necessary to pass argument to dependencies?

### ISSUE THREAD

**Issue by itzMeerkat** [tiangolo/fastapi#1206]
### First check

* [:heavy_check_mark:] I used the GitHub search to find a similar issue and didn't find it.
* [✔️] I searched the FastAPI documentation, with the integrated search.
* [✔️] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description
As title.
For example, I want to give different API different access restrictions (like permission level). My authentication is separated from the application so what I have are uid and corresponding user information. 
Now I want to use `Depends(check_token_and_permission)` to check if a certain request is valid. It seems not possible to pass an extra argument to the dependency function. I've read through the document and realized we have a workaround by creating a dependency class. However, I think it's less flexible than if we can do 
`Depends(check_token_and_permission, 'admin')`
and the function will be called as
`check_token_and_permission(a,b,c,*args)`
 where our argument 'admin' goes into *args

My question is, is that possible? Or is my reasoning correct? If not, why current implementation is better?

Thank you so much for answering my question.

**Comment by phy25** [comment]
Have you read https://fastapi.tiangolo.com/advanced/advanced-dependencies/#parameterized-dependencies?

**Comment by itzMeerkat** [comment]
> Have you read https://fastapi.tiangolo.com/advanced/advanced-dependencies/#parameterized-dependencies?

Yes, the "dependency class" actually referring to this method. Sorry for the confusion.

**Comment by acnebs** [comment]
I might be wrong, but I think the reason to use the class-based approach and instances of those classes instead of passing in args each time is that it's much easier to *cache* the dependencies, which is a very useful feature.

**Comment by itzMeerkat** [comment]
Well, that works for me, just curious XD.
Thank you all for answering!

**Comment by tiangolo** [comment]
Thanks for the help here @phy25 and @acnebs ! :cake: :rocket: 

Yeah, @itzMeerkat that would be the way to do it. On the other side, if you are dealing with security / OAuth2 scopes you can use `Security` with scopes, and it will be integrated into the OpenAPI / UI, etc.

https://fastapi.tiangolo.com/advanced/security/oauth2-scopes/

Anyway, thanks for reporting back and closing the issue! :+1:
